core/ROB.cpp

```diff
@@ -250,54 +249,26 @@ namespace olympia
         }
     }
 
-    // sys instr doesn't have a pipe so we handle special stuff here
+    // sys gets flushed unless it is csr rd
     void ROB::retireSysInst_(InstPtr &ex_inst)
     {
-        auto reg_file = ex_inst->getRenameData().getDestination().rf;
-        if (reg_file == core_types::RegFile::RF_INVALID)
-        {   // this is the case if dst = x0
-            DLOG("retiring SYS instr ");
-        } else   // this is needed or else destination register is 
-        {        // forever reserved and not ready
-            const auto & dest_bits = ex_inst->getDestRegisterBitMask(reg_file);
-            scoreboard_views_[reg_file]->setReady(dest_bits);
-            DLOG("retiring SYS inst dest reg bits: " <<  sparta::printBitSet(dest_bits));
-        }
-
-        FlushManager::FlushingCriteria criteria(FlushManager::FlushCause::POST_SYNC, ex_inst);
+        auto srclist = ex_inst->getRenameData().getSourceList();
+        // if SYS instr is not a csr instruction flush
+        // if SYS instr is a csr but src1 is not x0, flush
+        // otherwise it is a csr read, therefore don't flush
+        if (ex_inst->getMnemonic().substr(0,3)!="csr" || srclist.size()!=0)
```

Comment:
String comparisons are slower, you can use Mavis info to check for csr:
`ex_inst->getOpCodeInfo()->isInstType(mavis::InstMetaData::InstructionTypes::CSR)`

Reply:
@ah-condor -- can you make this change?

Reply:
Working on it!